### PR TITLE
Add import export buttons

### PR DESCRIPTION
**Pull request by anthonydillon**

### CODE REVIEW

**Comment by anthonydillon** [comment]
## QA
- Create a bundle
- Click export and see it exports
- Start a fresh canvas
- Click import and see that a file picker opens
- Click on the bundle.yaml and upload
- Check the upload blue bars show and are removed when the bundle appears
- Check the drag and drop functionality still works

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4114/
Test PASSed.

File: jujugui/static/gui/src/app/app.js [hunk 1]
```diff
@@ -1063,9 +1063,15 @@ YUI.add('juju-gui', function(Y) {
       a user drags a yaml file or zip file over the canvas.
 
       @method _renderDragOverNotification
+      @param {Boolean} showIndicator
     */
-    _renderDragOverNotification: function() {
-      this.views.environment.instance.fadeHelpIndicator(true);
+    _renderDragOverNotification: function(showIndicator) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/app.js]: Lets use a little ES6 here to make this nicer:
```JavaScript
_renderDragOverNotification: function(showIndicator = true) {
  this.views.environment.instance.fadeHelpIndicator(showIndicator);
  ...
}
```

File: jujugui/static/gui/src/app/components/deployment-bar/deployment-bar.js [hunk 2]
```diff
@@ -77,12 +77,35 @@ YUI.add('deployment-bar', function() {
...
+    },
+
+    /**
+      Open a file picker when the button is clicked.
+
+      @method _handleImport
+    */
+    _handleImportFile: function() {
+      var input = this.refs['file-input'];
+      app._renderDragOverNotification(false);
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-bar/deployment-bar.js]: This method needs to be passed in as a prop, you aren't 'allowed' to access things not passed in.

File: jujugui/static/gui/src/app/components/deployment-bar/deployment-bar.js [hunk 3]
```diff
@@ -77,12 +77,35 @@ YUI.add('deployment-bar', function() {
...
+      app.bundleImporter.importBundleFile(input.files[0]);
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-bar/deployment-bar.js]: to prop

File: jujugui/static/gui/src/app/components/deployment-bar/deployment-bar.js [hunk 4]
```diff
@@ -77,12 +77,35 @@ YUI.add('deployment-bar', function() {
...
+      setTimeout(() => {app._hideDragOverNotification();}, 600);
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-bar/deployment-bar.js]: to prop

File: jujugui/static/gui/src/app/components/deployment-bar/test-deployment-bar.js [hunk 5]
```diff
@@ -62,6 +62,16 @@ describe('DeploymentBar', function() {
           tabIndex="0">
           Export
         </span>
+        <span className="deployment-bar__import link"
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-bar/test-deployment-bar.js]: You'll need to add a test which checks to see that your new functionality works.

- Stub the props
- Set the value of the input field
- Simulate a click on the span
- Check that the props were called with the proper data.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4116/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4117/
Test FAILed.

File: jujugui/static/gui/src/app/components/deployment-bar/test-deployment-bar.js [hunk 6]
```diff
@@ -107,6 +117,21 @@ describe('DeploymentBar', function() {
     assert.equal(exportEnvironmentFile.callCount, 1);
   });
 
+  fit('can import an env', function() {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-bar/test-deployment-bar.js]: only this single test ran :) plz remove 'f'

File: jujugui/static/gui/src/app/components/deployment-bar/test-deployment-bar.js [hunk 7]
```diff
@@ -107,6 +117,21 @@ describe('DeploymentBar', function() {
...
+    var currentChangeSet = {one: 1, two: 2};
+    var renderDragOverNotification = sinon.stub();
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/deployment-bar/test-deployment-bar.js]: not used, did you forget an assertion?

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4118/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4120/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4121/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4122/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4123/
Test PASSed.

**Comment by hatched** [comment]
:+1: :ok: ! Thanks this looks awesome

File: jujugui/static/gui/src/app/assets/css/_header.scss [hunk 8]
```diff
@@ -35,7 +35,7 @@
             position: relative;
             left: -5px;
             top: 1px;
-            color: #cdcdcd;
+            color: $mid-grey;
         }
```
Comment (kadams54) [inline jujugui/static/gui/src/app/assets/css/_header.scss]: Nice cleanup.

**Comment by kadams54** [comment]
:+1: LGTM. Thanks!

**Comment by anthonydillon** [comment]
Thanks all :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**